### PR TITLE
GET topics added 

### PR DESCRIPTION
**Pull request by Karumu05**

### CODE REVIEW

File: __tests__/mainTestSuite.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            })
+            expect(body.topics[0]['slug']).toBe('mitch')
+        })
+    });
+    it('should respond with a status code of 200', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+    });
+    xit('GET:404 sends an appropriate status and response message when there is nothing in the data base' , () => {
```
Comment (nc-mezz-davies) [inline __tests__/mainTestSuite.test.js:34]: It took me a while to work out what you were trying to achieve here but you don't ned to worry - you can assume there will always be topics in the DB. Please do not skip tests though. You are free to delete this one!

File: app/app.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const express = require('express');
+const app = express();
+const {
+    getTopics,
+} = require('./controllers/topics')
+
+app.use(express.json());
```
Comment (nc-mezz-davies) [inline app/app.js:7]: Please only add code that is required for use in the branch. This is not required by your app/server as of yet so shouldn't be added. (You have no endpoints that use the request object.)

File: app/app.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+app.get('/api/topics', getTopics)
+
+
+app.use((err, request, response, next) => {
+
+    if(err.status && err.msg){
```
Comment (nc-mezz-davies) [inline app/app.js:14]: This too is unneeded code as there are no errors from the client side to deal with (I think this is part of your 404 for an empty database test right?)

File: app/models/topics.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const db = require('../../db/connection');
+
+exports.selectTopics = () => {
+    return db.query('SELECT * FROM topics').then((result) => {
+        if (result.rows.length === 0){
+            return Promise.reject({status: 404, msg: 'No topics exist'})
```
Comment (nc-mezz-davies) [inline app/models/topics.js:6]: No need to custom handle as per my other comments. Please delete this part of the code.

**Review by nc-mezz-davies — COMMENTED** [review]
I think you have got a little confused with the possible 404 error at this point. The only error at this point that would be a 404 is if the user tried to go to an endpoint that **doesn't exist.** 

You can certainly test for this and return a 404 status along with a message of your choosing but it is **not** specific to this endpoint - more the server itself - a generic 404. I hope my other comments clear this up further.

File: app/app.js (hunk 3 above)
Comment (Karumu05) [reply app/app.js:14]: yes it is for the 404 test

File: __tests__/mainTestSuite.test.js (hunk 1 above)
Comment (Karumu05) [reply __tests__/mainTestSuite.test.js:34]: Okay, I will I think i got hung up on error testing too quickly

**Pull request merged**